front/lib/api/assistant/actions/retrieval.ts

```diff
@@ -586,64 +589,33 @@ export class RetrievalConfigurationServerRunner extends BaseActionConfigurationS
 
           const refs = getRefs().slice(refsOffset, refsOffset + topK);
 
-          documents = v.map((d, i) => {
-            const reference = refs[i % refs.length];
-            return {
-              id: 0, // dummy pending database insertion
-              dataSourceWorkspaceId:
-                dataSourcesIdToWorkspaceId[d.data_source_id],
-              dataSourceId: d.data_source_id,
-              documentId: d.document_id,
-              reference,
-              timestamp: d.timestamp,
-              tags: d.tags,
-              sourceUrl: d.source_url ?? null,
-              score: d.chunks.map((c) => c.score)[0],
-              chunks: d.chunks.map((c) => ({
-                text: c.text,
-                offset: c.offset,
-                score: c.score,
-              })),
-            };
-          });
-        }
-      }
-    }
-
-    // We are done, store documents and chunks in database and yield the final events.
-
-    await frontSequelize.transaction(async (t) => {
-      for (const d of documents) {
-        const document = await RetrievalDocument.create(
-          {
-            dataSourceWorkspaceId: d.dataSourceWorkspaceId,
-            dataSourceId: d.dataSourceId,
-            sourceUrl: d.sourceUrl,
-            documentId: d.documentId,
-            reference: d.reference,
-            documentTimestamp: new Date(d.timestamp),
-            tags: d.tags,
-            score: d.score,
-            retrievalActionId: action.id,
-          },
-          { transaction: t }
-        );
-
-        d.id = document.id;
-
-        for (const c of d.chunks) {
-          await RetrievalDocumentChunk.create(
-            {
-              text: c.text,
-              offset: c.offset,
-              score: c.score,
-              retrievalDocumentId: document.id,
-            },
-            { transaction: t }
+          documents = await Promise.all(
```

Comment:
Was the transaction equivalent to `Promise.all` ? I would imagine the tx only takes a single connection no ?

Reply:
I think we want to avoid doing Promise.al

Reply:
Since the transaction now lives in the `makeNew` it's just that we will create many transactions. Happy to bring the `concurrentExecutor` in `front` if we really want to avoid `Promise.all`, let me know what you think!

Reply:
Can't we pass the transaction to makeNew ? Not having the tx feels worse than what we had previously

Reply:
Ok seen IRL.